src/common/include/macros.fpp

```diff
@@ -117,3 +117,19 @@
 
 #define t_vec3   real(kind(0d0)), dimension(1:3)
 #define t_mat4x4 real(kind(0d0)), dimension(1:4,1:4)
+
+#:def ASSERT(predicate, message = None)
+    if (.not. (${predicate}$)) then
+        call s_mpi_abort("${_FILE_.split('/')[-1]}$:${_LINE_}$: "// &
+                         "Assertion failed: ${predicate}$. " &
+                         //${message or '"No error description."'}$)
+    end if
+#:enddef
+
+#:def PROHIBIT(predicate, message = None)
```

Comment:
same as the assert comment

Reply:
Same as my answer for `@:PROHIBIT`. Assert is just the opposite of prohibit. Depending on the condition, it can clearer to use one or the other.